adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,40 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
+#include <algorithm>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    std::stack<char> bracketStack;
+
+    for (auto ch : source)
+    {
+        for (auto br : brackets)
```

Comment:
please avoid this nested loop, use it in appropriate context for searching with algorithms